From 3b9cd8161adc37b595ab92d02b4bcf342d5e081a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kacper=20Michaj=C5=82ow?= <kasper93@gmail.com>
Date: Fri, 9 Jan 2026 00:43:22 +0100
Subject: [PATCH] meson: allow building against glslang from VULKAN_SDK

Meson doesn't support searching for glslang from VULKAN_SDK, also it
doesn't give access to env variables, so add an option. Note that for
vulkan dependency Meson automatically uses VULKAN_SDK.
---
 meson_options.txt    |  3 +++
 src/glsl/meson.build | 19 ++++++++++++-------
 2 files changed, 15 insertions(+), 7 deletions(-)

diff --git a/meson_options.txt b/meson_options.txt
index 5e582fe5e..95a96c73f 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -8,6 +8,9 @@ option('vk-proc-addr', type: 'feature', value: 'auto',
 option('vulkan-registry', type: 'string', value: '',
        description: 'Path to vulkan XML registry (for code generation)')
 
+option('vulkan-sdk', type: 'string', value: '',
+       description: 'Path to Vulkan SDK (for glslang)')
+
 option('opengl', type: 'feature', value: 'auto',
        description: 'OpenGL-based renderer')
 
diff --git a/src/glsl/meson.build b/src/glsl/meson.build
index b1f4c8d09..c9a85429d 100644
--- a/src/glsl/meson.build
+++ b/src/glsl/meson.build
@@ -38,13 +38,18 @@ elif not glslang_req.disabled()
   # TODO: Use the following once it's fixed
   # glslang = dependency('glslang', method: 'cmake', modules: ['glslang::SPIRV'])
 
+  vulkan_sdk = get_option('vulkan-sdk')
+  vulkan_lib_dirs = []
+  if vulkan_sdk != ''
+    vulkan_lib_dirs += [vulkan_sdk / 'lib']
+  endif
   prefer_static = get_option('prefer_static')
   found_lib = false
   foreach arg : [[prefer_static, false], [not prefer_static, glslang_req]]
     static   = arg[0]
     required = arg[1]
 
-    spirv = cxx.find_library('SPIRV', required: required, static: static)
+    spirv = cxx.find_library('SPIRV', required: required, static: static, dirs: vulkan_lib_dirs)
 
     if not spirv.found()
       continue
@@ -59,13 +64,13 @@ elif not glslang_req.disabled()
     if static
       glslang_deps += [
         # Always required for static linking
-        cxx.find_library('MachineIndependent', required: false, static: true),
-        cxx.find_library('OSDependent',        required: false, static: true),
-        cxx.find_library('OGLCompiler',        required: false, static: true),
-        cxx.find_library('GenericCodeGen',     required: false, static: true),
+        cxx.find_library('MachineIndependent', required: false, static: true, dirs: vulkan_lib_dirs),
+        cxx.find_library('OSDependent',        required: false, static: true, dirs: vulkan_lib_dirs),
+        cxx.find_library('OGLCompiler',        required: false, static: true, dirs: vulkan_lib_dirs),
+        cxx.find_library('GenericCodeGen',     required: false, static: true, dirs: vulkan_lib_dirs),
         # SPIRV-Tools are required only if optimizer is enabled in glslang build
-        cxx.find_library('SPIRV-Tools',        required: false, static: true),
-        cxx.find_library('SPIRV-Tools-opt',    required: false, static: true),
+        cxx.find_library('SPIRV-Tools',        required: false, static: true, dirs: vulkan_lib_dirs),
+        cxx.find_library('SPIRV-Tools-opt',    required: false, static: true, dirs: vulkan_lib_dirs),
       ]
     endif
 
